ignite/engine/__init__.py

```diff
@@ -109,44 +111,36 @@ def supervised_training_step_amp(
             tuple of tensors `(batch_x, batch_y)`.
         output_transform (callable, optional): function that receives 'x', 'y', 'y_pred', 'loss' and returns value
             to be assigned to engine's state.output after each iteration. Default is returning `loss.item()`.
-        scaler (torch.cuda.amp.GradScaler, bool, optional): GradScaler instance for gradient scaling.
-            If True, will create default GradScaler. If GradScaler instance is passed, it will be used instead.
-            (default: False)
+        scaler (torch.cuda.amp.GradScaler, optional): GradScaler instance for gradient scaling. (default: None)
 
     Returns:
         Tuple[Callable, Optional[torch.cuda.amp.GradScaler]]: update function and GradScaler instance
+
+    .. versionadded:: 0.5.0
     """
 
     try:
-        from torch.cuda import amp
+        from torch.cuda.amp import autocast
     except ModuleNotFoundError:
         raise ModuleNotFoundError("Please install torch>=1.6.0 to use amp_mode='amp'.")
 
-    if scaler and isinstance(scaler, bool):
-        scaler_ = amp.GradScaler(enabled=True)
-    elif isinstance(scaler, amp.GradScaler):
-        scaler_ = scaler
-
     def update(engine: Engine, batch: Sequence[torch.Tensor]) -> Union[Any, Tuple[torch.Tensor]]:
         model.train()
         optimizer.zero_grad()
         x, y = prepare_batch(batch, device=device, non_blocking=non_blocking)
-        with amp.autocast(enabled=True):
+        with autocast(enabled=True):
             y_pred = model(x)
             loss = loss_fn(y_pred, y)
-        if scaler_:
-            scaler_.scale(loss).backward()
-            scaler_.step(optimizer)
-            scaler_.update()
+        if scaler:
+            scaler.scale(loss).backward()
+            scaler.step(optimizer)
+            scaler.update()
         else:
             loss.backward()
             optimizer.step()
         return output_transform(x, y, y_pred, loss)
 
-    if scaler:
-        return update, scaler_
-    else:
-        return update, None
+    return update, scaler
```

Comment:
```suggestion
    return update
```

Return scaler is useless now